src/Internal/Types/ScriptLookups.purs

```diff
@@ -1161,14 +1211,21 @@ processConstraint mpsMap osMap = do
             }
         _cpsToTxBody <<< _outputs %= Array.(:) txOut
         _valueSpentBalancesOutputs <>= provideValue amount
-    MustPayToScript vlh dat datp scriptRef plutusValue -> do
+    MustPayToScript vlh mbCredential dat datp scriptRef plutusValue -> do
```

Comment:
Should `MustSpendScriptOutput` be updated as well? since we can now pay to a base address but `MustSpendScriptOutput` still expects an enterprise address.

Could at least add a Plutip test to see if we can spend from a base address.

Reply:
Yep, sounds right.

Reply:
I updated `MustSpendScriptOutput` handler to also take base addresses into account.